internal/backend/runtime/omni/controllers/omni/internal/task/machine/poll.go

```diff
@@ -327,6 +332,28 @@ func pollMeta(ctx context.Context, c *client.Client, info *Info) error {
 		})
 }
 
+func detectOverlay(ctx context.Context, c *client.Client) (*schematic.Overlay, error) {
+	reader, err := c.Read(ctx, "/proc/cmdline")
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := io.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	cmdline := procfs.NewCmdline(string(data))
+
+	value := cmdline.Get(constants.KernelParamBoard)
+
+	if value == nil || value.Get(0) == nil {
```

Comment:
1.7+ doesn't set the `talos.board` args, so after upgrading how's it going to handle it?

Reply:
it'll get overlay from the schematic info in extraData